lib/oban/plugins/pruner.ex

```diff
@@ -78,9 +97,27 @@ defmodule Oban.Plugins.Pruner do
 
   @impl GenServer
   def handle_info(:prune, %State{} = state) do
-    Query.with_xact_lock(state.conf, state.lock_key, fn ->
-      delete_jobs(state.conf, state.max_age, state.limit)
-    end)
+    start_metadata = %{config: state.conf, plugin: __MODULE__}
+
+    :telemetry.span(
+      [:oban, :plugin],
+      start_metadata,
+      fn ->
+        state.conf
+        |> Query.with_xact_lock(state.lock_key, fn ->
+          delete_jobs(state.conf, state.max_age, state.limit)
+        end)
+        |> case do
+          {:ok, {jobs_pruned_count, _}} ->
+            metadata = %{jobs_pruned_count: jobs_pruned_count}
+            {:ok, Map.merge(start_metadata, metadata)}
+
+          error ->
+            metadata = %{error: error}
+            {:error, Map.merge(start_metadata, metadata)}
+        end
+      end
```

Comment:
This will return `false` if another process has the advisory lock already. That's an expected situation and shouldn't fall through to the error case. The nested functions and `case` piping seems a little awkward to me, perhaps we can make a helper function to `lock_and_delete_jobs` in this case?

```suggestion
      fn ->
        case lock_and_delete_jobs(state) do
          {:ok, {count, _}} ->
            {:ok, Map.put(start_metadata, :pruned_count, count)}

          false ->
            {:ok, Map.put(start_metadata, :pruned_count, 0)}

          error ->
            {:error, Map.merge(start_metadata, :error, error)}
        end
      end
```